frontend/src/metabase/admin/databases/containers/DatabaseEditApp.unit.spec.js

```diff
@@ -34,46 +37,16 @@ jest.mock(
   () => ComponentMock,
 );
 
-function mockSettings({ cachingEnabled = false }) {
-  const original = MetabaseSettings.get.bind(MetabaseSettings);
-  const spy = jest.spyOn(MetabaseSettings, "get");
-  spy.mockImplementation(key => {
-    if (key === "engines") {
-      return ENGINES_MOCK;
-    }
-    if (key === "enable-query-caching") {
-      return cachingEnabled;
-    }
-    if (key === "site-url") {
-      return "http://localhost:3333";
-    }
-    if (key === "application-name") {
-      return "Metabase Test";
-    }
-    if (key === "is-hosted?") {
-      return false;
-    }
-    if (key === "cloud-gateway-ips") {
-      return [];
-    }
-    return original(key);
-  });
-}
-
 async function setup({ cachingEnabled = false } = {}) {
-  mockSettings({ cachingEnabled });
-
-  const settingsReducer = () => ({
-    values: {
-      engines: ENGINES_MOCK,
-      "enable-query-caching": cachingEnabled,
-      "persisted-models-enabled": false,
-    },
+  const settings = mockSettings({
+    engines: ENGINES_MOCK,
+    "enable-query-caching": cachingEnabled,
+    "persisted-models-enabled": false,
```

Comment:
How do you determine what settings to keep here, since the previous version mock 6 settings keys and we only keep 3 here.

Reply:
Earlier `MetabaseSettings` wasn't receiving any values when running tests and they had to be set/mocked manually. The settings we've removed here were used by some components deep down the tree and weren't really relevant to these tests. Now when we're using `mockSettings`, both redux store and `MetabaseSettings` are populated with a full settings object (with some meaningful defaults). So we don't need to e.g. set `site-url` to something just because some component we don't care about needs it. `mockSettings` just sets `site-url` to some URL and does the same with other possible settings.

Reply:
Ahhh. Got it. The default is from calling `createMockSettings` inside `mockSettings`. This makes sense 👍